launcher/minecraft/mod/tasks/LocalModParseTask.cpp

```diff
@@ -62,6 +62,36 @@ ModDetails ReadMCModInfo(QByteArray contents)
         for (auto author : authors) {
             details.authors.append(author.toString());
         }
+
+        if (details.mod_id.startsWith("mod_")) {
+            details.mod_id = details.mod_id.mid(4);
+        }
+
+        auto addDep = [&details](QString dep) {
+            if (dep == "mod_MinecraftForge" || dep == "Forge")
+                return;
+            if (dep.contains(":")) {
+                dep = dep.section(":", 1);
+            }
+            if (dep.contains("@")) {
+                dep = dep.section("@", 0, 0);
+            }
+            if (dep.startsWith("mod_")) {
+                dep = dep.mid(4);
+            }
+            details.dependencies.append(dep);
+        };
+
+        if (firstObj.contains("requiredMods")) {
+            for (auto dep : firstObj.value("dependencies").toArray().toVariantList()) {
+                addDep(dep.toString());
+            }
+        } else if (firstObj.contains("dependencies")) {
+            for (auto dep : firstObj.value("dependencies").toArray().toVariantList()) {
+                addDep(dep.toString());
+            }
+        }
```

Comment:
nitpick: just use json api directly
but you did also accidentally write dependencies in the first case :P
```suggestion
        if (firstObj.contains("requiredMods")) {
            for (auto dep : firstObj.value("requiredMods").toArray()) {
                addDep(dep.toString());
            }
        } else if (firstObj.contains("dependencies")) {
            for (auto dep : firstObj.value("dependencies").toArray()) {
                addDep(dep.toString());
            }
        }
```